Newsletters: add 301 (2024-05-08)

harding opened the pull request:
- [x] Finish first draft @harding 
- [x] Lede, releases/RCs, topic entries @harding 
- [x] PR Review Club section @stickies-v 

Sorry this is late.  I got nerd snipped by the lamport discussion.  ~~I'll take this out of draft when the main content is ready for review.~~  Ready for review now; sorry for the delay!

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
````diff
@@ -0,0 +1,261 @@
+  ```
+
+  In binary, that's 10101, or 21 in decimal.  Bob verifies each of the
+  preimages matches the public key he previously received, assuring him
````
```suggestion
  preimages match the public keys he previously received, assuring him
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  generated the message "21".
+
+  For ECDSA signatures, Bitcoin uses the [DER encoding standard][],
+  which omits leading zero (0x00) bytes from the the signature's two
```
```suggestion
  which omits leading zero (0x00) bytes from the signature's two
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  time, so Bitcoin signatures naturally vary in size.  This variation is
+  exacerbated by the leading byte for R values being a 0x00 half the
+  time (see [low-r grinding][topic low-r grinding]) but, in theory, the
+  variation can be reduced a transaction being one byte smaller 1/256th
```
```suggestion
  variation can be reduced by a transaction being one byte smaller 1/256th
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  that transaction will still need to include any additional data
+  necessary to make it valid, such as hash preimages.
+
+  This allows the P2SH redeem script to contains an ECDSA signature
```
```suggestion
  This allows the P2SH redeem script to contain an ECDSA signature
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  kept secret.  However, after the lamport signature is published, an
+  attacker who wants to reuse it with a forged ECDSA signature will have
+  to ensure the ECDSA signature is the same length as the original ECDSA
+  signature.  This may require the attacker grind the signature,
```
```suggestion
  signature.  This may require the attacker to grind the signature,
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  they must reveal the preimage for `<digest_y>`.  If a valid preimage
+  for the actual signature size isn't revealed, the script fails.
+
+  Now even if ECDSA is completely broken, it's not possible for an
+  attacker to spend the bitcoins unless they also know the lamport
```
```suggestion
  Now even if ECDSA is completely broken, an
  attacker can't spend the bitcoins unless they also know the lamport
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  lamport signatures.  Unfortunately, because ECDSA signatures only
+  naturally vary in byte size one time out of every 256, the
+  straightforward way to do this would require a very large number of
+  signatures to be included in order to obtain practical security.
```
```suggestion
  signatures to be included to obtain practical security.
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  is not known, so anyone trying to avoid this attack would be unable to
+  spend their own bitcoins until a fast quantum computer was invented.
+
+  This lamport signature verification is practically similar to the
```
```suggestion
  This lamport signature verification is similar to the
```

harding replied:
Leaving as-is.  I want to convey that both accomplish the same thing but that they differ in how they accomplish it.  I also want to leave some wiggle room in case the differences are larger than I expect from my initial reading of the idea.  :-)

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+
+  This lamport signature verification is practically similar to the
+  proposed [OP_CHECKSIGFROMSTACK][topic op_checksigfromstack] opcode.
+  In both cases the data to be verified, a public key, and a signature
```
```suggestion
  In both cases, the data to be verified, a public key, and a signature
```

LarryRuane commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,261 @@
+  proposed [OP_CHECKSIGFROMSTACK][topic op_checksigfromstack] opcode.
+  In both cases the data to be verified, a public key, and a signature
+  are placed on the stack and an operation only succeeds if the
+  signature both corresponds to public key and commits to the data on
```
```suggestion
  signature both corresponds to the public key and commits to the data on
```

stickies-v commented:
Added review club commit - will take another look tomorrow with a set of fresh eyes.

jirijakes commented on _posts/en/newsletters/2024-05-08-newsletter.md:
````diff
@@ -0,0 +1,315 @@
+  key:
+
+  ```text
+  public_zeroes = [private_zeroes[0], ..., private_zeroes[4]]
+  public_ones   = [private_ones[0], ..., private_ones[4]]
````
```suggestion
  public_zeroes = [hash(private_zeroes[0]), ..., hash(private_zeroes[4])]
  public_ones   = [hash(private_ones[0]), ..., hash(private_ones[4])]
```

harding replied:
Oof, thanks so much for this!  I guess this is why I mainly write documentation, not code!  :-)

jirijakes commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,315 @@
+  shared between the signer and the validator interprets which preimages
+  are revealed as instructions.  For example, Bob wants to verify that
+  Alice signed a number between 0 and 31 (in binary, 00000 to 11111).
+  Alice private creates a lamport private key from two lists of random
```
```suggestion
  Alice creates a lamport private key from two lists of random
```

harding commented:
Added remaining core content and addressed all feedback to date (thanks @LarryRuane @jirijakes!).  I'll be back in a couple hours to add the remaining bits and to review @stickies-v's section (thanks!).

bitschmidty commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,311 @@
+
+  It's known that this mechanism can be defeated by an attacker.
+  Bitcoin Core's orphan pool is a circular buffer that can be added to
+  by all of its peers, so an attacker who want to prevent this type of
```
```suggestion
  by all of its peers, so an attacker who wants to prevent this type of
```

bitschmidty commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,311 @@
+  the default configuration to also use DNS seeds, they would both be
+  contacted in parallel and whichever was faster would dominate the peer
+  addresses that the node would try.  Given the lower overhead of DNS
+  and that fact that results might already be cached by a server
```
```suggestion
  and the fact that results might already be cached by a server
```

bitschmidty commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,311 @@
+
+- [Bitcoin Core #28016][] begins waiting for all seed nodes to be polled
+  before polling DNS seeds.  Users can configure both seed nodes and DNS
+  seeds.  A seed node is a regular Bitcoin full node; Bitcoin Core can
```
IIRC, It might be worth distinguishing in the writeup that DNS seed are hard coded and no seed nodes are hard coded. A point you somewhat touch on in the last sentence.

bitschmidty commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,311 @@
+
+- [Bitcoin Core #29623][] makes various improvements to warning users if
+  their local time seems to be more than 10 minutes out of sync with the
+  time of their connected peers.  A node with a bad clock might
```
Think referencing this PR could be helpful for readers not already familiar with it:
https://bitcoinops.org/en/newsletters/2024/02/07/#bitcoin-core-28956

harding commented:
Made edits for @bitschmidty follow-up (thanks!), made some own edits of my own (including correcting the described protocol---oops!), added lede, added releases/RCs, and added topic entries.  Reviewed and added one link to @stickies-v section (great job, thanks!).

azuchi commented on _posts/en/newsletters/2024-05-08-newsletter.md:
```diff
@@ -0,0 +1,336 @@
+  Bitcoin-related cryptographic operations.  It speeds up key generation
+  and signing (see [last week's newsletter][news300 secp]) and reduces
+  the compiled size "which we expect to benefit embedded users in
+  particular."  It also adds a function for sorting pulic keys.
```
```suggestion
  particular."  It also adds a function for sorting public keys.
```

harding replied:
Fixed, thanks @azuchi!

bitschmidty reviewed (commented):
I pushed a small fixup for harding's text, ACK the topics, lede and other updates.
